src/asr/decoders/greedy_ctc_decoder.cpp

```
@@ -49,39 +49,44 @@ CtcHeadModule::build_graph(
 
     // (T, num_classes+1, 1, 1) -> (num_classes+1, T, 1, 1)
     auto logits_t = ggml_cont(bf_ctx.ctx, ggml_permute(bf_ctx.ctx, logits.tensor, 1, 0, 2, 3));
-    auto sm = ggml_soft_max(bf_ctx.ctx, logits_t);
-    auto log_sm = ggml_log(bf_ctx.ctx, sm);
+    return ggml_runtime::ggml_bf_tensor(ggml_soft_max(bf_ctx.ctx, logits_t), x.buft);
+}
+
+ggml_runtime::TensorBag
+CtcHeadModule::build_graph(
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)